acceptancetests/reporting.py

```diff
@@ -0,0 +1,90 @@
+"""Reporting helper class for communicating with influx db."""
+import abc
+import time
+import datetime
+try:
+    import urlparse
+except ImportError:
+    import urllib.parse as urlparse
+
+from influxdb import InfluxDBClient
+from influxdb.client import InfluxDBClientError
+
+__metaclass__ = type
+
+DBNAME = 'txn_metrics'
+POLICYNAME = 'txn_metric'
+
+class _Reporting:
+    """_Reporting represents a class to report metrics upon"""
+
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, client):
+        self.client = client
+        self.labels = {
+            "total_time": "txn_metric.total_time",
+            "total_num_txns": "txn_metric.total_num_txns",
+            "max_time": "txn_metric.max_time",
+            "test_duration": "txn_metric.test_duration",
+        }
+
+    @abc.abstractmethod
+    def report(self, metrics, tags):
+        """Report the metrics to the underlying reporting client
+        """
+
+class InfluxDB(_Reporting):
+    """InfluxDB represents a influx db reporting client"""
+
+    def __init__(self, *args, **kwargs):
+        super(InfluxDB, self).__init__(*args, **kwargs)
+
+    def report(self, metrics, tags):
+        now = datetime.datetime.today()
+        series = []
+        for key, label in self.labels:
+            if key in metrics:
+                pointValue = {
+                    "measurement": label,
+                    "tags": tags,
+                    "time": int(now.strftime('%s')),
+                    "fields": {
+                        "value": metrics[key],
+                    },
+                }
+                series.append(pointValue)
+        self.client.write_points(series, retention_policy=POLICYNAME)
+
+def makeMetrics(total_time, total_num_txns, max_time, test_duration):
```

Comment:
this should just be make_metrics if it's just a function in the package.

Reply:
So this is called by `./assess_deploy_webscale.py` - so that we don't have to keep track of the metric labels outside of the `reporting` module.

Reply:
right, I just mean python standards would be function names would be written with underscores.

Reply:
Ah yes, so it should, I changed it! You probably want to code review this whole folder - :see_no_evil: